airflow/utils/setup_teardown.py

```diff
@@ -110,25 +110,49 @@ def push_setup_teardown_task(cls, operator: Operator | list[Operator]):
             if first_task.is_teardown:
                 if not all(task.is_teardown == first_task.is_teardown for task in operator):
                     raise ValueError("All tasks in the list must be either setup or teardown tasks")
+                upstream_tasks = [task for task in first_task.upstream_list]
```

Comment:
```suggestion
                upstream_tasks = first_task.upstream_list
```

No need to iterate it here. It's already a list. Might not be worth having another variable and just keep using `first_task.upstream_list`? (only commenting this once)

Reply:
There is another in the `else` block as well